### PR TITLE
Change type of PointIterator in PointEvalHandler.jl

### PR DESCRIPTION
**Pull request by suurj**
Change the type of the coordinate vector to a more general type in PointIterator. The old type works only if the spatial dimension of the coordinates is 2.

### CODE REVIEW

File: src/PointEval/PointEvalHandler.jl [hunk 1]
```diff
@@ -297,9 +297,9 @@ end
 """
 PointIterator
 
-struct PointIterator{PH<:PointEvalHandler}
+struct PointIterator{PH<:PointEvalHandler, V <: Vector{<: Vec}}
```
Comment (fredrikekre) [inline src/PointEval/PointEvalHandler.jl:300]:
```suggestion
struct PointIterator{PH<:PointEvalHandler, V <: Vec}
```

File: src/PointEval/PointEvalHandler.jl [hunk 2]
```diff
@@ -297,9 +297,9 @@ end
...
     ph::PH
-    coords::Vector{Vec{2,Float64}}
+    coords::V
```
Comment (fredrikekre) [inline src/PointEval/PointEvalHandler.jl:302]:
```suggestion
    coords::Vector{V}
```

**Review by fredrikekre — COMMENTED** [review]
Thanks! Thats a silly bug left over from testing most likely. Would you be able to add a simple test case for this in the `test/test_pointevaluation.jl` file?

**Comment by suurj** [comment]
> Thanks! Thats a silly bug left over from testing most likely. Would you be able to add a simple test case for this in the `test/test_pointevaluation.jl` file?

I changed the `PointIterator` test in `test/test_pointevaluation.jl` from two-dimensional to three-dimensional. It seems to work fine.

File: test/test_pointevaluation.jl [hunk 3]
```diff
@@ -129,10 +129,10 @@ end
 function dofhandler2()
     # Computes the L2 projection of a quadratic field exactly
     # but not using L2Projector since we want the DofHandler dofs
-    mesh = generate_grid(Quadrilateral, (20, 20))
-    ip_f = Lagrange{RefQuadrilateral,2}()
-    ip_f_v = ip_f^2
-    qr = QuadratureRule{RefQuadrilateral}(3)
+    mesh = generate_grid(Hexahedron, (10, 10, 10))
```
Comment (fredrikekre) [inline test/test_pointevaluation.jl:132]: We should still test 2D as well. Perhaps you can make the dimension an input variable to the function (`dofhandler2`) and call it once for 2D and once for 3D?

Comment (suurj) [reply test/test_pointevaluation.jl]: Thanks for the suggestion. I added a boolean parameter into `dofhandler2` to test both cases.

**Comment by fredrikekre** [comment]
Thanks for the fix, and welcome as a contributor!

**Pull request merged**